test/core/security/grpc_audit_logging_test.cc

```diff
@@ -134,36 +126,38 @@ TEST(StdoutLoggerTest, StdoutLoggerCreationAndLogInvocation) {
       AuditLoggerRegistry::CreateAuditLogger(std::move(result.value()));
   AuditContext context("method", "spiffe", "policy", "rule", true);
   ::testing::internal::CaptureStdout();
+  int64_t time_before_log = absl::ToUnixSeconds(absl::Now());
   logger->Log(context);
+  int64_t time_after_log = absl::ToUnixSeconds(absl::Now());
   auto log_or = JsonParse(::testing::internal::GetCapturedStdout());
   ASSERT_TRUE(log_or.ok());
-  ASSERT_EQ(log_or.value().type(), Json::Type::kObject);
-  auto it = log_or.value().object().find("grpc_audit_log");
-  ASSERT_NE(it, log_or.value().object().end());
+  ASSERT_EQ(log_or->type(), Json::Type::kObject);
+  auto it = log_or->object().find("grpc_audit_log");
+  ASSERT_NE(it, log_or->object().end());
   ASSERT_EQ(it->second.type(), Json::Type::kObject);
-  auto object = it->second.object();
-  // Check if the recorded timestamp is close to now.
-  absl::Time t1 = absl::Now();
+  auto& object = it->second.object();
   ASSERT_NE(object.find("timestamp"), object.end());
-  EXPECT_EQ(object.find("timestamp")->second.type(), Json::Type::kNumber);
-  uint64_t ts;
-  ASSERT_TRUE(absl::SimpleAtoi(object.find("timestamp")->second.string(), &ts));
-  absl::Time t2 = absl::FromUnixSeconds(ts);
-  EXPECT_TRUE(t1 - t2 < absl::Seconds(10));
+  EXPECT_EQ(object.find("timestamp")->second.type(), Json::Type::kString);
+  absl::Time log_time;
+  ASSERT_TRUE(absl::ParseTime(absl::RFC3339_sec,
+                              object.find("timestamp")->second.string(),
+                              &log_time, nullptr));
+  int64_t time_at_log = absl::ToUnixSeconds(log_time);
+  // Check if the recorded timestamp is in between the recorded interval with
+  // the precision of one second.
+  EXPECT_TRUE(time_at_log >= time_before_log && time_at_log <= time_after_log);
   // Check exact values of everything else.
-  std::map<std::string, std::string> fields = {{"rpc_method", "method"},
-                                               {"principal", "spiffe"},
-                                               {"policy_name", "policy"},
-                                               {"matched_rule", "rule"}};
-  for (const auto& p : fields) {
-    auto it = object.find(p.first);
+  std::vector<std::string> fields = {"rpc_method", "principal", "policy_name",
```

Comment:
This should use `absl::string_view` instead of `std::string` to avoid unnecessary copying.

But actually, better yet would be to just exclude the one field that we *don't* want to check, so that we don't need to remember to add field names here if we ever add new fields to the output format:

```
Json::Object json_object = object;
json_object.erase("timestamp");
```